### ISSUE TITLE
Absurd altitude variation while it could be read from .gpx track file

### ISSUE THREAD

**Issue by faegan** [PokemonGoF/PokemonGo-Bot#4560]
### Expected Behavior

The bot reads information from the .gpx track file, including altitude. When moving towards a waypoint, the "moving from" altitude should always match the altitude of the provided "to" location OR get altitude information from google maps.

### Actual Behavior

The bot generates a random altitude between alt_min and alt_max and varies that every step taken and passes it as "from" location. This results in absurd altitude variation within 2 seconds.

### Your FULL config.json (remove your username, password, gmapkey and any other private info)

http://pastebin.com/k74tJBss

### Output when issue occurred

http://pastebin.com/96r5fiTN

### Steps to Reproduce

fill in alt_min = 500 and alt_max = 1000 in config.json (these are default values)

### Other Information
OS: Windows 10
 
Branch: DEV
 
Git Commit: 8276a3330725142299c64a61513408d6dd2ed22c
 
Python Version: Python 2.7.12
 

Any other relevant files/configs (eg: path files)

**Comment by faegan** [comment]
Also, the following is silly for The Netherlands:
pokecli.py: error: --alt_min is out of range! (should be >= 0.0)

Most of our country is < 0.0

**Comment by mjmadsen** [comment]
Sorry about this @faegan. The initial altitude changes were added a bit willy nilly so we would stop sending 0 at all times. I can alter the alt_min value to better represent land altitudes. 

As far as why other portions are not grabbing the alt when available, I'm not sure. Current work around is alter the min/max range to somewhere proportional to the area in which you are in.

**Comment by faegan** [comment]
Thank you @mjmadsen I've changed it to represent somewhat valid altitudes. I'd still like to see it use known altitude information from the track path file. I do not know if all .gpx files have this, but I used an export from gpsies and it contains this:

`<gpsies:minHeightMeter>-7.0</gpsies:minHeightMeter>
      <gpsies:maxHeightMeter>1.0</gpsies:maxHeightMeter>`

Could we override alt_min and alt_max when this information is in the path/track file?

**Comment by mjmadsen** [comment]
commit 4411b93 for your min_alt problem. 

I haven't looked at those files too much yet. I'll put it on the list. Ideally, gpsies should rule everything except noise/jitter - I don't think we're close to that yet.

**Comment by mjmadsen** [comment]
While what is up is not perfect, at least you can put a min max for the area you're actually trying to bot in. I've been working on debugging pokemon catch bits of late, as well as misc. bugs. I'm hoping someone who is more familiar with those particular files will see this and put up a PR.

**Comment by faegan** [comment]
it's not to do with the files, I've narrowed it down to the StepWalker;

`self.alt = uniform(self.bot.config.alt_min, self.bot.config.alt_max)`

it seems it generates an altitude for each step instead of accepting it from the call in follow_path.py.

**Comment by mjmadsen** [comment]
Ah. Probably need to add in an alt parameter with a default value. If we're at default, we generate random. Something like that...

**Comment by faegan** [comment]
I kinda made it work for myself now, but I'll leave the coding to others.
I have edited the following files to make it work:
follow_path.py if statement in line 141:
`step_walker = StepWalker(
                self.bot,
                lat,
                lng
            )`

changed to:

`step_walker = StepWalker(
                self.bot,
                lat,
                lng,
		alt
            )`

and step_walker.py line 10:

`def __init__(self, bot, dest_lat, dest_lng, dest_alt=None):
        self.bot = bot
        self.api = bot.api

        self.initLat, self.initLng = self.bot.position[0:2]

        self.dist = distance(
            self.initLat,
            self.initLng,
            dest_lat,
            dest_lng
        )
		
        self.alt = dest_alt
        if dest_alt is None:
            self.alt = uniform(self.bot.config.alt_min, self.bot.config.alt_max)`

**Comment by mjmadsen** [comment]
That's great. Would you want to take a look at the other follow_ files and make the same changes and put up a PR?

**Comment by faegan** [comment]
@mjmadsen I have no idea how this works :)

**Comment by mjmadsen** [comment]
To be you pretty much did it in your comment! While it won't fix all issues, passing the alt grabbed from each walker and having our stepwalker accept and apply it is the first step.

**Comment by mjmadsen** [comment]
Does #4581 satisfy what you're looking for?

**Comment by mjmadsen** [comment]
Closing, if there is an issue after #4599, it's probably with a specific walker that we haven't updated to send/get alt (new issue).
